### PR TITLE
Visibility required

### PR DESCRIPTION
**Pull request by MathiasReker**
Visibility MUST be declared on all properties and methods; abstract and final MUST be declared before the visibility; static MUST be declared after the visibility.

### CODE REVIEW

File: lib/PhpParser/Lexer/TokenEmulator/FlexibleDocStringEmulator.php [hunk 1]
```diff
@@ -6,7 +6,7 @@
 
 final class FlexibleDocStringEmulator extends TokenEmulator
 {
-    const FLEXIBLE_DOC_STRING_REGEX = <<<'REGEX'
+    public const FLEXIBLE_DOC_STRING_REGEX = <<<'REGEX'
```
Comment (nikic) [inline lib/PhpParser/Lexer/TokenEmulator/FlexibleDocStringEmulator.php:9]: Can be private.

File: lib/PhpParser/Lexer/TokenEmulator/NumericLiteralSeparatorEmulator.php [hunk 2]
```diff
@@ -7,13 +7,13 @@
...
-    const EXP = '(?:e[+-]?' . self::DEC . ')';
-    const FLOAT = '(?:' . self::SIMPLE_FLOAT . self::EXP . '?|' . self::DEC . self::EXP . ')';
-    const NUMBER = '~' . self::FLOAT . '|' . self::BIN . '|' . self::HEX . '|' . self::DEC . '~iA';
+    public const BIN = '(?:0b[01]+(?:_[01]+)*)';
+    public const HEX = '(?:0x[0-9a-f]+(?:_[0-9a-f]+)*)';
+    public const DEC = '(?:[0-9]+(?:_[0-9]+)*)';
+    public const SIMPLE_FLOAT = '(?:' . self::DEC . '\.' . self::DEC . '?|\.' . self::DEC . ')';
+    public const EXP = '(?:e[+-]?' . self::DEC . ')';
+    public const FLOAT = '(?:' . self::SIMPLE_FLOAT . self::EXP . '?|' . self::DEC . self::EXP . ')';
+    public const NUMBER = '~' . self::FLOAT . '|' . self::BIN . '|' . self::HEX . '|' . self::DEC . '~iA';
```
Comment (nikic) [inline lib/PhpParser/Lexer/TokenEmulator/NumericLiteralSeparatorEmulator.php:16]: These can be private.

File: lib/PhpParser/Parser/Php7.php [hunk 3]
```diff
@@ -17,146 +17,146 @@
...
+    public const T_PAAMAYIM_NEKUDOTAYIM = 386;
+    public const T_NAMESPACE = 387;
+    public const T_NS_C = 388;
+    public const T_DIR = 389;
+    public const T_NS_SEPARATOR = 390;
+    public const T_ELLIPSIS = 391;
+    public const T_NAME_FULLY_QUALIFIED = 392;
+    public const T_NAME_QUALIFIED = 393;
+    public const T_NAME_RELATIVE = 394;
+    public const T_ATTRIBUTE = 395;
```
Comment (nikic) [inline lib/PhpParser/Parser/Php7.php:159]: These are generated files, you'd have to modify the template file.

File: lib/PhpParser/ParserAbstract.php [hunk 4]
```diff
@@ -27,7 +27,7 @@
 
 abstract class ParserAbstract implements Parser
 {
-    const SYMBOL_NONE = -1;
+    public const SYMBOL_NONE = -1;
```
Comment (nikic) [inline lib/PhpParser/ParserAbstract.php:30]: This can probably be private.

File: lib/PhpParser/PrettyPrinterAbstract.php [hunk 5]
```diff
@@ -14,13 +14,13 @@
...
-    const FIXUP_BRACED_NAME     = 4; // Name operand that may require bracing
-    const FIXUP_VAR_BRACED_NAME = 5; // Name operand that may require ${} bracing
-    const FIXUP_ENCAPSED        = 6; // Encapsed string part
+    public const FIXUP_PREC_LEFT       = 0; // LHS operand affected by precedence
+    public const FIXUP_PREC_RIGHT      = 1; // RHS operand affected by precedence
+    public const FIXUP_CALL_LHS        = 2; // LHS of call
+    public const FIXUP_DEREF_LHS       = 3; // LHS of dereferencing operation
+    public const FIXUP_BRACED_NAME     = 4; // Name operand that may require bracing
+    public const FIXUP_VAR_BRACED_NAME = 5; // Name operand that may require ${} bracing
+    public const FIXUP_ENCAPSED        = 6; // Encapsed string part
```
Comment (nikic) [inline lib/PhpParser/PrettyPrinterAbstract.php:23]: These can be protected.

File: lib/PhpParser/Lexer/TokenEmulator/NumericLiteralSeparatorEmulator.php (hunk 2 above)
Comment (MathiasReker) [reply lib/PhpParser/Lexer/TokenEmulator/NumericLiteralSeparatorEmulator.php]: fixed

File: lib/PhpParser/ParserAbstract.php (hunk 4 above)
Comment (MathiasReker) [reply lib/PhpParser/ParserAbstract.php]: fixed

File: lib/PhpParser/PrettyPrinterAbstract.php (hunk 5 above)
Comment (MathiasReker) [reply lib/PhpParser/PrettyPrinterAbstract.php]: fixed

File: lib/PhpParser/Parser/Php7.php (hunk 3 above)
Comment (MathiasReker) [reply lib/PhpParser/Parser/Php7.php:159]: I understand the file will be overridden. Should I revert this change or keep or keep it as it is?

File: lib/PhpParser/PrettyPrinterAbstract.php [hunk 6]
```diff
@@ -14,13 +14,13 @@
...
-    const FIXUP_PREC_LEFT       = 0; // LHS operand affected by precedence
-    const FIXUP_PREC_RIGHT      = 1; // RHS operand affected by precedence
-    const FIXUP_CALL_LHS        = 2; // LHS of call
-    const FIXUP_DEREF_LHS       = 3; // LHS of dereferencing operation
-    const FIXUP_BRACED_NAME     = 4; // Name operand that may require bracing
-    const FIXUP_VAR_BRACED_NAME = 5; // Name operand that may require ${} bracing
-    const FIXUP_ENCAPSED        = 6; // Encapsed string part
+    public const FIXUP_PREC_LEFT          = 0; // LHS operand affected by precedence
+    public const FIXUP_PREC_RIGHT         = 1; // RHS operand affected by precedence
+    public const FIXUP_CALL_LHS           = 2; // LHS of call
```
Comment (nikic) [inline lib/PhpParser/PrettyPrinterAbstract.php:19]: Why are these three left as public?

File: lib/PhpParser/Parser/Php7.php (hunk 3 above)
Comment (nikic) [reply lib/PhpParser/Parser/Php7.php:159]: You can keep this change if you also update this line: https://github.com/nikic/PHP-Parser/blob/de4ac93023427c778eecc5ce14d194f1b33c8cd6/grammar/parser.template#L27

File: lib/PhpParser/Lexer/TokenEmulator/NumericLiteralSeparatorEmulator.php (hunk 2 above)
Comment (nikic) [reply lib/PhpParser/Lexer/TokenEmulator/NumericLiteralSeparatorEmulator.php:16]: It looks like these are still public.

**Pull request closed without merging**